src/clientRPC/ClientClient.ts

```diff
@@ -0,0 +1,345 @@
+import type { ReadableWritablePair } from 'stream/web';
+import type { TLSSocket } from 'tls';
+import type { NodeId } from 'ids/index';
+import { WritableStream, ReadableStream } from 'stream/web';
+import { createDestroy } from '@matrixai/async-init';
+import Logger from '@matrixai/logger';
+import WebSocket from 'ws';
+import { PromiseCancellable } from '@matrixai/async-cancellable';
+import { Timer } from '@matrixai/timer';
+import { Validator } from 'ip-num';
+import * as clientRpcUtils from './utils';
+import * as clientRPCErrors from './errors';
+import { promise } from '../utils';
+
+const timeoutSymbol = Symbol('TimedOutSymbol');
+
+interface ClientClient extends createDestroy.CreateDestroy {}
+@createDestroy.CreateDestroy()
+class ClientClient {
```

Comment:
This matches `QUICClient`, but it should be renamed `WebSocketClient`.

Both clients are `CreateDestroy` patterns.

Reply:
This is done.